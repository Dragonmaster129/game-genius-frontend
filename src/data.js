const data = {
  player: 3,
  profession: "Doctor",
  auditor: "Sam",
  savings: 400,
  assets: {
    salary: 13200,
    interest: [
      // { name: "thingy", value: 40, key: 1 },
      // { name: "notherone", value: 50, key: 2 },
      // { name: "notherone", value: 50, key: 3 },
      // { name: "notherone", value: 50, key: 4 },
      // { name: "notherone", value: 50, key: 5 },
    ],
    dividends: [
      // { name: "2Big", value: 30, key: 1 }
    ],
    realestate: [
      // {
      //   type: "3/2",
      //   name: "3/2 House",
      //   cost: 55000,
      //   downpay: 5000,
      //   value: 200,
      //   key: 1,
      // },
      // {
      //   type: "3/2",
      //   name: "3/2 House",
      //   cost: 55000,
      //   downpay: 7000,
      //   value: 400,
      //   key: 2,
      // },
    ],
    businesses: [
      // {
      //   type: "autoParts",
      //   name: "Auto Parts",
      //   cost: 100000,
      //   downpay: 100000,
      //   value: 1500,
      //   key: 1,
      // },
      // {
      //   type: "autoParts",
      //   name: "Auto Parts",
      //   cost: 120000,
      //   downpay: 120000,
      //   value: 1700,
      //   key: 2,
      // },
    ],
    stock: [
      // { name: "OK4U", amount: 1000, costPerShare: 1, key: 1 }
    ],
  },
  expenses: {
    taxes: 3420,
    mortgage: { monthly: 1900, totalCost: 202000 },
    school: { monthly: 750, totalCost: 150000 },
    car: { monthly: 380, totalCost: 19000 },
    creditCard: { monthly: 270, totalCost: 9000 },
    retail: { monthly: 50, totalCost: 1000 },
    other: 2880,
    child: {
      count: 0,
      costPer: 640,
    },
    loan: 0,
  },
};

export default data;
