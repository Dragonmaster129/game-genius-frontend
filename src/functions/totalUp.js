const totalUp = (objToTotal) => {
  let total = 0;
  if (Object.prototype.toString.call(objToTotal) === "[object Object]") {
    for (const key in objToTotal) {
      let element = objToTotal[key];
      if (key != "passive" && key != "loan") {
        if (Array.isArray(element)) {
          for (let index = 0; index < element.length; index++) {
            const item = element[index];
            if (item.value) {
              total = total + item.value;
            }
          }
        } else if (
          Object.prototype.toString.call(element) === "[object Object]"
        ) {
          if (element.monthly) {
            total = total + element.monthly;
          } else {
            total = total + element.count * element.costPer;
          }
        } else {
          total = total + element;
        }
      } else if (key == "loan") {
        total = total + element / 10;
      }
    }
  } else if (Array.isArray(objToTotal)) {
    for (let index = 0; index < objToTotal.length; index++) {
      const element = objToTotal[index];
      total = total + element.value;
    }
  }

  return total;
};

export default totalUp;
