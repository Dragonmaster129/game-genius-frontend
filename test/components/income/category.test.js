import React from "react";
import renderer from "react-test-renderer";
import Category from "../../../src/components/income/category";
import data from "../../sampledata";

import totalUp from "../../../src/functions/totalUp";

let interestIsOpen = false;

const showData = (dataArr) => {
  return dataArr.map((item) => {
    return (
      <div className="item hz" key={item.key}>
        <div>{item.name}:</div>
        <div>{item.value}</div>
      </div>
    );
  });
};

test("Component returns a value", () => {
  const component = renderer.create(
    <Category
      showData={showData}
      items={data.assets.interest}
      isOpen={interestIsOpen}
      click={() => (interestIsOpen = !interestIsOpen)}
      className="interest"
      title="Interest"
      totalValue={totalUp(data.assets.interest)}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
