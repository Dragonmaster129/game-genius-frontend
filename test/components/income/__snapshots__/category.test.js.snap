// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component returns a value 1`] = `
<div
  className="interest category"
>
  <h3
    className="category-click"
    onClick={[Function]}
  >
    Interest:240
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-down "
      data-icon="angle-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 384c-8.188 0-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L192 306.8l137.4-137.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-160 160C208.4 380.9 200.2 384 192 384z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h3>
  <div
    className="item-wrapper hidden"
  >
    <div
      className="item hz"
    >
      <div>
        thingy
        :
      </div>
      <div>
        40
      </div>
    </div>
    <div
      className="item hz"
    >
      <div>
        notherone
        :
      </div>
      <div>
        50
      </div>
    </div>
    <div
      className="item hz"
    >
      <div>
        notherone
        :
      </div>
      <div>
        50
      </div>
    </div>
    <div
      className="item hz"
    >
      <div>
        notherone
        :
      </div>
      <div>
        50
      </div>
    </div>
    <div
      className="item hz"
    >
      <div>
        notherone
        :
      </div>
      <div>
        50
      </div>
    </div>
  </div>
</div>
`;
